eng/install-native-dependencies.sh

```diff
@@ -13,37 +15,35 @@
 #
 # ./install-native-dependencies.sh <OS>
 
-if [ "$1" = "Linux" ]; then
-    sudo apt update
-    if [ "$?" != "0" ]; then
-       exit 1;
-    fi
-    sudo apt install cmake llvm-3.9 clang-3.9 lldb-3.9 liblldb-3.9-dev libunwind8 libunwind8-dev gettext libicu-dev liblttng-ust-dev libcurl4-openssl-dev libssl-dev libkrb5-dev libnuma-dev build-essential
-    if [ "$?" != "0" ]; then
-        exit 1;
-    fi
-elif [[ "$1" == "MacCatalyst" || "$1" == "OSX" || "$1" == "tvOS" || "$1" == "iOS" ]]; then
-    engdir=$(dirname "${BASH_SOURCE[0]}")
+os="$(echo "$1" | tr "[:upper:]" "[:lower:]")"
+
+if [ -e /etc/os-release ]; then
+    . /etc/os-release
+fi
 
-    echo "Installed xcode version: `xcode-select -p`"
+if [ "$os" = "linux" ] && { [ "$ID" = "debian" ] || [ "$ID_LIKE" = "debian" ]; }; then
+    apt update
+
+    apt install -y build-essential gettext locales cmake llvm clang lldb liblldb-dev libunwind8-dev libicu-dev liblttng-ust-dev \
+        libssl-dev libkrb5-dev libnuma-dev
+
+    localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
```

Comment:
This wasn't there before, right? Why is this required?

Reply:
Yes, before locales and libz-dev were missing and libcurl-openssl-dev was unnecessary. So it was simply outdated.

e.g. given a new environment:

```sh
docker run --rm -v $(pwd):/runtime ubuntu sh -c \
  '/runtime/eng/install-native-dependencies.sh linux && /runtime/build.sh libs.native'
```

without any locales gives a lot of errors like this:

```
Build succeeded.

bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/runtime/src/native/libs/build-native.proj]
bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/runtime/src/native/libs/build-native.proj]
bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/runtime/src/native/libs/build-native.proj]
bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/runtime/src/native/libs/build-native.proj]
bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/runtime/src/native/libs/build-native.proj]
bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/runtime/src/native/libs/build-native.proj]
bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/runtime/src/native/libs/build-native.proj]
bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/runtime/src/native/libs/build-native.proj]
bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/runtime/src/native/libs/build-native.proj]
bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/runtime/src/native/libs/build-native.proj]
bash : error : setlocale: LC_ALL: cannot change locale (en_US.UTF-8) [/runtime/src/native/libs/build-native.proj]
    0 Warning(s)
    11 Error(s)

Time Elapsed 00:02:57.08
Build failed with exit code 1. Check errors above.
```

Reply:
Looks good to me, it is what we use in some Docker setup scripts like https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/db4fcc31b3fe5315bc0ae7f02e9e34cd16bbe102/src/debian/11/gcc12/Dockerfile#L24

And we're only using this script on macOS so it doesn't impact our builds anyway.